File: exam_solutions/examA/index_of.py
def index_of(values, value):
    i = 0
    for current_value in values:
        if current_value == value:
            return i
        i = i + 1
    return -1

if __name__ == "__main__":
    values = [0, 1, 2, 3, 4, 5, 6, 7]
    value = 3
    print("The index of {} in {} is {}".format(value, values, index_of(values, value)))

File: exam_solutions/examA/index_of.cpp
#include <iostream>

template <typename T>
int index_of(const int size, T values[], T elem, bool (*equals)(T a, T b));

bool equals_int(int a, int b) {
    return a == b;
}

int main() {
    int size = 10;
    int pvalues[size];
    for (int i = 0; i < size; i++) {
        pvalues[i] = i;
    }
    int value_to_search = 7;
    std::cout << "Index of pointer to 7 is : " << index_of(size, pvalues, value_to_search, equals_int) << std::endl;
    return 0;
}

template <typename T>
int index_of(const int size, T values[], T elem, bool (*equals)(T a, T b)) {
    for (int i = 0; i < size; i++) {
        if (equals(values[i], elem)) {
            return i;
        }
    }
    return -1;
}